sdk/translation/Azure.AI.Translation.Text/src/Custom/TextTranslationTranslateOptions.cs

```diff
@@ -0,0 +1,200 @@
+        /// looking up supported languages using the translation scope. If the from parameter isn&apos;t specified,
+        /// automatic language detection is applied to determine the source language.
+        ///
+        /// You must use the from parameter rather than autodetection when using the dynamic dictionary feature.
+        /// Note: the dynamic dictionary feature is case-sensitive.
+        /// </summary>
+        public string SourceLanguage { get; set; }
+        /// <summary>
+        /// Defines whether the text being translated is plain text or HTML text. Any HTML needs to be a well-formed,
+        /// complete element. Possible values are: plain (default) or html.
+        /// </summary>
+        public TextType? TextType { get; set; }
+        /// <summary>
+        /// A string specifying the category (domain) of the translation. This parameter is used to get translations
+        /// from a customized system built with Custom Translator. Add the Category ID from your Custom Translator
+        /// project details to this parameter to use your deployed customized system. Default value is: general.
+        /// </summary>
+        public string Category { get; set; }
+        /// <summary>
+        /// Specifies how profanities should be treated in translations.
+        /// Possible values are: NoAction (default), Marked or Deleted.
+        /// </summary>
+        public ProfanityAction? ProfanityAction { get; set; }
+        /// <summary>
+        /// Specifies how profanities should be marked in translations.
+        /// Possible values are: Asterisk (default) or Tag.
+        /// </summary>
+        public ProfanityMarker? ProfanityMarker { get; set; }
+        /// <summary>
+        /// Specifies whether to include alignment projection from source text to translated text.
+        /// Possible values are: true or false (default).
+        /// </summary>
+        public bool? IncludeAlignment { get; set; }
+        /// <summary>
+        /// Specifies whether to include sentence boundaries for the input text and the translated text.
+        /// Possible values are: true or false (default).
+        /// </summary>
+        public bool? IncludeSentenceLength { get; set; }
+        /// <summary>
+        /// Specifies a fallback language if the language of the input text can&apos;t be identified.
+        /// Language autodetection is applied when the from parameter is omitted. If detection fails,
+        /// the SuggestedFrom language will be assumed.
+        /// </summary>
+        public string SuggestedFrom { get; set; }
+        /// <summary>
+        /// Specifies the script of the input text.
+        /// </summary>
+        public string FromScript { get; set; }
+        /// <summary>
+        /// Specifies the script of the translated text.
+        /// </summary>
+        public string ToScript { get; set; }
```

Comment:
nit: If we have "SourceLanguage" and "TargetLanguage", should these be called "SourceScript" and "TargetScript"? Could be a good question to discuss during an architecture review.

Reply:
Agreed.  We will update these names in accordance with our architecture review.  Model properties in general are to be updated in this task:

This change will fall under this task to update the model properties.:

https://github.com/Azure/azure-sdk-for-net/issues/35669